Packs/DigitalShadows/Integrations/ReliaQuestGreyMatterDRPIncidents/ReliaQuestGreyMatterDRPIncidents.py

```diff
@@ -0,0 +1,1011 @@
+                                                        event_created_after=sinceDate,
+                                                        alert_risk_types=riskTypes, risk_level=riskLevel,
+                                                        should_ingest_closed=ingestClosed)
+    data = poll_result.triage_data
+    last_polled_number = poll_result.max_event_number
+    if data:
+        incidents = []
+        for item in data:
+            incident = {
+                'name': item["triage_item"]['title'],
+                'occurred': item["triage_item"]['raised'],
+                'rawJSON': json.dumps(item)
+            }
+            incidents.append(incident)
+        demisto.incidents(incidents)
+        LOG(f"settings  last run: {last_event_num}")
+        demisto.setLastRun({'incidents': {'last_fetch': last_polled_number}})
+        LOG(f"data found for iteration last_polled_number:{last_polled_number}")
+    else:
+        LOG(f"No data found for iteration last_polled_number:{last_polled_number}")
+        demisto.incidents([])
+        LOG(f"settings  last run: {last_event_num}")
+        demisto.setLastRun({'incidents': {'last_fetch': last_polled_number}})
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    """
+        PARSE AND VALIDATE INTEGRATION PARAMS
+    """
+    LOG(f'input config------: {demisto.params()}')
+    secretKey = demisto.params()["apiSecret"]['password']
+    accessKey = demisto.params()['apiKey']['password']
+    accountId = demisto.params()['accountId']
+    riskTypes = demisto.params()['riskTypes']
+    riskLevel = demisto.params()['riskLevel']
+    ingestClosed = demisto.params().get('ingestClosedIncidents')
+
+    if RISK_TYPE_ALL in riskTypes:
+        riskTypes = [RISK_TYPE_ALL]
+    if RISK_LEVEL_ALL in riskLevel:
+        riskLevel = [RISK_LEVEL_ALL]
+
+    # get the service API url
+    base_url = demisto.params()['searchLightUrl']
+    verify_certificate = not demisto.params().get('insecure', False)
+    fetchLimit = arg_to_number(demisto.params()['fetchLimit'], "fetchLimit", True)
+    if fetchLimit > 100:
+        raise DemistoException("fetch limit must be less than 100")
+    sinceDate = parse_date(demisto.params()['sinceDate'])
+    demisto.params().get('proxy', False)
+    LOG(f'Command being called is {demisto.command()}')
+    try:
+        search_light_client = Client(
+            base_url=base_url,
+            account_id=accountId,
+            access_key=accessKey,
+            secret_key=secretKey,
+            verify=verify_certificate
+        )
+        # Creating different client as base url is different here which pointing to old apis for ds-search command
+        digital_shadows_client = Client(
+            base_url=DS_BASE_URL,
+            account_id=accountId,
+            access_key=accessKey,
+            secret_key=secretKey,
+            verify=verify_certificate
+        )
```

Comment:
Why do we need to instantiate 2 clients? I see that you're using `search_light_client` for `test-module` and `fetch-incidents` but `digital_shadows_client` is used in `ds-search`. Remember that this code is only run once and it will hit one of the `demisto.command == $COMMAND` leafs. So in the end, only 1 client will actually be used. 

I think a better approach is creating some helper function that will determine what `base_url` is depending on which command is run and then just create the client that is needed for that command.

```suggestion
        base_url = get_base_url(command: str = demisto.command())
        digital_shadows_client = Client(
            base_url=DS_BASE_URL,
            account_id=accountId,
            access_key=accessKey,
            secret_key=secretKey,
            verify=verify_certificate
        )
        
        if demisto.command() == 'test-module':
            # This is the call made when pressing the integration Test button.
            result = test_module(client)
        elif demisto.command() == 'fetch-incidents':
            result = fetch_incidents(fetchLimit, ingestClosed, riskLevel, riskTypes, client, sinceDate)
        elif demisto.command() == 'ds-search':
            result = search_find(client, demisto.args())
        else:
            raise NotImplementedError(f'{demisto.command()} command is not implemented.')
            
        return_result(result)
```

Reply:
In case of fetch_incidents result will not initialised and give error.